### PR TITLE
Api resource docs and test

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary

Adds tests and docs for API Resource.

Opening as WIP. Mostly to remind myself that I haven't finished this yet.

## TODO

- [x] Have tests been written for the new code?
- [x] Has documentation been written/updated?

### CODE REVIEW

**Comment by DXCanas** [comment]
Is this jasmine syntax?

**Comment by rtibbles** [comment]
Mocha.

**Comment by rtibbles** [comment]
This is as much as I will get done on this.

File: kolibri/core/assets/src/api-resource.js [hunk 1]
```diff
@@ -322,6 +322,23 @@ class Collection {
...
+    return JSON.stringify(
+      Object.assign(
+        {}, ...Object.keys(params).sort().map(
+          (paramKey) => ({ [paramKey]: params[paramKey] })
+        )
+      )
+    );
+  }
+
+  get key() {
```
Comment (ralphiee22) [inline kolibri/core/assets/src/api-resource.js:340]: To clear up my understanding, when we ask for `collection.key` that calls this getter method, which in turn calls the static method to produce the key?

**Comment by indirectlylit** [comment]
If this is a WIP, reviewing with intent to merge?

**Comment by ralphiee22** [comment]
I think it is no longer a WIP. I have already reviewed.

File: kolibri/core/assets/src/api-resource.js (hunk 1 above)
Comment (rtibbles) [reply kolibri/core/assets/src/api-resource.js:340]: Yes.

**Pull request merged**